examples/check/check.bal

```diff
@@ -12,12 +12,14 @@ type Address record {
 
 function getAddress(Person p) returns (Address|error) {
     // If the address does not exist, return an error.
-    match (p.address) {
-        Address add => { return add;}
-        () => {
-            error addNotFoundErr = error("address not found");
-            return addNotFoundErr;
-        }
+    // For type-guard to work, only simple variable references should be used. therefore, `p.address` is assigned to a
+    // variable.
```

Comment:
Shall we make this explanation a bit more comprehensive? i.e., the difference between type guard and type test and maybe demonstrate both cases

Reply:
done